### DISCUSSION TITLE
When will you provide support for github merge queues?

### DISCUSSION THREAD

**Discussion by jabalsad** [Mergifyio/mergify#5092] [General]
Hello,

We use mergify to automate some of our github workflows. As of today, it doesn't support taking the "add to merge queue" action for Github merge queues. Do you have plans to support it?

**Comment by DouglasBlackwood** [comment]
Hi,

We can't provide support at this moment as GitHub doesn't provide any API to interact with their merge queue.
Though, we do have our own merge queue solution, we thus invite you to go check our documentation at: https://docs.mergify.com/actions/queue/

**Comment by jabalsad** [comment]
That's not entirely true. We enqueue PRs to the merge queue using their graphql API:

```
      - name: Add PR to merge queue
        env:
          GITHUB_TOKEN: ${{ github.token }}
          PR_URL: ${{ github.event.pull_request.html_url }}
          PR_NODE_ID: ${{ github.event.pull_request.node_id }}
          MUTATION: |-
            mutation($pr: ID!) {
              enqueuePullRequest(input: {
                pullRequestId: $pr,
              }) { clientMutationId }
            }
        run: |
          # TODO use "merge pr" command when this is resolved:
          # https://github.com/cli/cli/issues/7213
          gh api graphql \
            -f query="$MUTATION" \
            -F pr="$PR_NODE_ID" \
```
